deploy/06_deploy_coverage_pool.js

```diff
@@ -0,0 +1,24 @@
+module.exports = async ({ getNamedAccounts, deployments }) => {
+  const { deploy } = deployments
+  const { deployer } = await getNamedAccounts()
+  const AssetPool = await deployments.get("AssetPool")
+  const UnderwriterToken = await deployments.get("UnderwriterToken")
+
+  const CoveragePool = await deploy("CoveragePool", {
+    from: deployer,
+    args: [AssetPool.address],
+    log: true,
+  })
+
+  const assetPool = await ethers.getContractAt("AssetPool", AssetPool.address)
+  const underwriterToken = await ethers.getContractAt(
+    "UnderwriterToken",
+    UnderwriterToken.address
+  )
+
+  await assetPool.transferOwnership(CoveragePool.address)
+  await underwriterToken.transferOwnership(AssetPool.address)
```

Comment:
The transactions will fail if we run the deployments scripts again e.g. to retry after some failure. We need to check the owner first.

```suggestion
  if ((await assetPool.owner()) !== CoveragePool.address) {
    await assetPool.transferOwnership(CoveragePool.address)
  }

  if ((await underwriterToken.owner()) !== AssetPool.address) {
    await underwriterToken.transferOwnership(AssetPool.address)
  }
```

Reply:
To be consistent with how we handle transaction execution in `11_approve_first_risk_manager.js` instead of `ethers` we could use what `deployments` have:
```js
  const { read, execute } = deployments
// ...
  if (
    (await read("AssetPool", { from: deployer }, "owner")) !==
    CoveragePool.address
  ) {
    await execute(
      "AssetPool",
      { from: deployer },
      "transferOwnership",
      CoveragePool.address
    )
  }

  if (
    (await read("UnderwriterToken", { from: deployer }, "owner")) !==
    AssetPool.address
  ) {
    await execute(
      "UnderwriterToken",
      { from: deployer },
      "transferOwnership",
      AssetPool.address
    )
  }
```

Reply:
503a296c6685a4796638fc7a988779dc6cd4ef28